src/cloudai/cli/handlers.py

```
@@ -132,64 +134,79 @@ def handle_dse_job(runner: Runner, args: argparse.Namespace) -> int:
+    # (a bug) is a hard-fail. We capture it so reports still generate, then re-raise below.
+    run_error: Exception | None = None
+    try:
+        for tr in runner.runner.test_scenario.test_runs:
+            test_run = copy.deepcopy(tr)
+
+            agent_type = test_run.test.agent
+            agent_class = registry.agents_map.get(agent_type)
+            if agent_class is None:
+                logging.error(
+                    f"No agent available for type: {agent_type}. Please make sure {agent_type} "
+                    f"is a valid agent type. Available agents: {registry.agents_map.keys()}"
```

Ruff (0.15.18)

[warning] Logging statement uses f-string

(G004)